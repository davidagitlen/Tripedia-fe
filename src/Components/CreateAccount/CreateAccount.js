import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import './CreateAccount.scss';
import banner from '../../images/banner.png'


const CreateAccount = () => {
  
  const [accountState, setAccountState] = useState({
    name: '',
    email: '',
    password: '',
    confirmPassword: '',
  })

  const {name, email, password, confirmPassword} = accountState
  const isEnabled = email && password && name && password && confirmPassword;
  
 const handleChange = (e) => {
   setAccountState({...accountState, [e.target.name]: e.target.value })
  }

  return(
    <main className='create-account__container'>
      <img src={banner} alt="Tripedia for all your travel planning needs" className="banner" />
      <form className="create-account__form">
        <input
          type="text"
          placeholder="name"
          name='name'
          value={name}
          onChange={handleChange}></input>
        <input
          type="email"
          name='email'
          placeholder="email"
          value={email}
          onChange={handleChange}>
        </input>
        <input
          type="password"
          name='password'
          placeholder="password"
          value={password}
          onChange={handleChange}>
        </input>
        <input
          type="password"
          name='confirmPassword'
          placeholder="confirm password"
          value={confirmPassword}
          onChange={handleChange}>
        </input>
        {!isEnabled && 
          <button className='button__disabled'
            type="submit">Submit
          </button>}
        {isEnabled &&
        <NavLink to='/'>
          <button className='button__enabled'
            type="submit">Submit
          </button>
        </NavLink>}
        <NavLink to="/login">
          <p>Login</p>
        </NavLink>
      </form>
    </main>
  )
}

export default CreateAccount;