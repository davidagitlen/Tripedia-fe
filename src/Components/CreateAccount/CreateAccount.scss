.create-account__container {
  padding-left: 125px;
  padding-right: 125px;
  padding-top: 100px;
  background-color: rgba(32, 113, 120, .3);
  height: 100vh;

  .banner {
    width: 100%;
    height: 350px;
    border-radius: 40px;
  }

  .create-account__form {
    height: 300px;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    padding-left: 150px;
    padding-right: 150px;
    margin-top: 15px;

    a {
      width: 100%;
      display: flex;
      justify-content: center;
      text-decoration: none;
    }
  }

  form {

    text-align: center;
  }

  input {
    width: 100%;
    height: 40px;
    border-radius: 40px;
    border: 2px solid grey;
    text-align: center;
  }

  .button__disabled {
    width: 50%;
    border-radius: 40px;
    height: 40px;
    background-color: rgba(128, 128, 128, .3)
  }

  .button__enabled {
    width: 50%;
    border-radius: 40px;
    height: 40px;
    background-color: #FF9766;
    color: #207178;
    font-weight: bold;
    font-size: 20px;
    border: 2px solid #207178;
  }
}

@media screen and (min-width: 1200px) {
  .create-account__container {
    padding-left: 400px;
    padding-right: 400px;
  }

  .create-account__container input,
  .create-account__container button {
    margin: 10px;
  }

  .create-account__container form {
    margin-top: 10px;
    padding-left: 350px;
    padding-right: 350px;
  }

}