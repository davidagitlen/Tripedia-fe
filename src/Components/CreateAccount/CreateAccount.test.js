import React from 'react';
import { shallow } from 'enzyme';
import CreateAccount from './CreateAccount';

describe('CreateAccount', () => {

  it('should pass a dummy test', () => {
    expect(true).toEqual(true);
  });

});