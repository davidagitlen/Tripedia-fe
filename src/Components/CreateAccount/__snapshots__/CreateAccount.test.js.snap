// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount should match the wrapper with data passed in 1`] = `
<main
  className="create-account__container"
>
  <img
    alt="Tripedia for all your travel planning needs"
    className="banner"
    src="banner.png"
  />
  <form
    className="create-account_form"
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="name"
      type="text"
      value=""
    />
    <input
      name="email"
      onChange={[Function]}
      placeholder="email"
      type="email"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
      value=""
    />
    <input
      name="password_confirmation"
      onChange={[Function]}
      placeholder="confirm password"
      type="password"
      value=""
    />
    <button
      className="button__disabled"
      type="submit"
    >
      Submit
    </button>
    <NavLink
      to="/"
    >
      <p>
        Login
      </p>
    </NavLink>
  </form>
</main>
`;
